examples/internal/proto/examplepb/a_bit_of_everything.swagger.json

```diff
@@ -2998,6 +3054,18 @@
         "int64OverrideType": {
           "type": "integer",
           "format": "int64"
+        },
+        "requiredStringViaFieldBehaviorAnnotation": {
+          "type": "string",
+          "title": "mark a field as required in Open API definition",
+          "required": [
+            "required_string_via_field_behavior_annotation"
```

Comment:
I believe this is a bug in the existing implementation as the other fields that are marked as required as also using snakecase for the field names here.

Reply:
This behavior is the same as today (see the `float_value` field) - but I don't believe it's actually correct. The OpenAPI spec is a little hard to read, but I think that what should happen here is that the `required` parameter should not be set on the field and instead the name of the field should go in the `required` array at the object level.

Reply:
Hm, we have generators that run against our generated files as an oracle for our output - with your suggested change do they respond correctly? Would you have time to fix this at the same time?

Reply:
If you take the OpenAPI spec that we are [generating](https://raw.githubusercontent.com/grpc-ecosystem/grpc-gateway/1f17ac3600b1bc59ecdc9e34a176ab9dd050deb6/examples/internal/proto/examplepb/a_bit_of_everything.swagger.json) and compare it against the [OpenAPI 2.0 JSON Schema](https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v2.0/schema.json), it [validates](https://www.jsonschemavalidator.net/) just fine. However, JSON Schema validation isn't strict - so the schema validation won't fail when the optional `required` field that is being added to `definitions` property.

What's not entirely clear to me in reading the specification is which of the fields for a [Schema Object](https://swagger.io/specification/v2/#schemaObject) are allowed at the top level vs. the field level. The specification says that these parameters derive from the [JSON Schema Validation](https://tools.ietf.org/html/draft-fge-json-schema-validation-00) spec which [clearly specifies](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.4.3) that `required` should be an array at an object level. Given that it does not specifically state that `required` can be used with numerics or strings, I think that answer is that `required` should be omitted there (but again doesn't hurt anything since it still passes validation).

I think there are 2 things that need to be fixed here:

1) When using camel case, the `required` field names need to be modified to account for it
2) My change is only setting `required` at the field level but is not adding it to the `required` array for the object which needs addressed

And this one is optional, but I think we can just ignore it:

3) The required property should not be set on the field definition for objects; as the required array (http://swagger.io/specification/#schemaObject) at the object level is actually responsible for setting this

Reply:
This is now addressed.